docs/DeveloperGuide.md

```diff
@@ -1,34 +1,247 @@
 # Developer Guide
 
-## Design & Implementation
+## 1. Setting Up
+   * Prerequisites
+     * JDK 11 or above
+     * IntelliJ IDE
+   * Setting up the project in your computer
+     * Fork this repo, and clone the fork to your computer
+     * Open IntelliJ (if you are not in the welcome screen, click File > Close Project to close the existing project dialog first)
+     * Set up the correct JDK version for Gradle
+     * Click Configure > Project Defaults > Project Structure
+     * Click New…​ and find the directory of the JDK
+     * Click Import Project
+     * Locate the build.gradle file and select it. Click OK
+     * Click Open as Project
+     * Click OK to accept the default settings.
+   * Verifying the setup
+     * Run the jikan.jikan and try a few commands
+     * Run the tests and ensure they all pass.
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+## 2. Design
+![image_info](./pictures/ClassDiagram.png)
+_Fig 2.1. Class diagram of the Jikan program_
 
 
-## Product Scope
-### Target user profile
+## 3. Implementation
 
-{Describe the target user profile}
+This section describes some noteworthy details on how certain features are implemented.
 
-### Value proposition
+### 3.1 Automated Storage Cleanup feature
 
-{Describe the value proposition: what problem does it solve?}
+#### 3.1.1 Current Implementation
 
-## User Stories
+The storage cleanup mechanism is stored internally as a StorageCleaner java class. The StorageCleaner class has an association with the Storage class and thus it is able to access and edit the datafile which contains the list of activities. 
+
+Additionally, when the StorageCleaner class is first initialised, it will create two files namely a status file and a data file in the “recycled” folder under the “data” folder. The status file keeps track of the activation status of the storage cleaner while the data file serves as a recycle bin for deleted data. 
+
+Moreover, the class also implements the following operations:
+
+`StorageCleaner#initialiseCleaner - Loads up the status file to decide whether to activate/deactivate the automated cleaner.`
+
+`StorageCleaner#setStatus - Gives the user the freedom to activate/de-activate the cleaner using commands during application runtime.`
+
+`StorageCleaner#autoClean - This operation is called whenever the application is executed. Storage cleaning will only be done if the cleaner is activated.`
+
+Given below is the example scenario of how the operations work.
+
+##### initialiseCleaner
+
+Step 1. Loads up the status file for reading. If the status file is not found, create a new status file and write the character ‘0’ to the first line of the status file.
+
+Step 2. Read the first line of the status file, if a character ‘0’ is found, deactivate the automated cleanup. Else if a character ‘1’ is found, activate the automated cleanup.
+
+![image_info](./pictures/FlowchartinitCleaner.png)
```

Comment:
The lines in the diagram are difficult to be seen. Perhaps it would be better to change the colour of the lines in the diagram so the diagram looks more visible for the reader?
Perhaps you could also consider adding in notes by the side of the diagram to indicate the meaning of "0" and "1" at the side of the diagram to make the diagram clearer?

Reply:
Alternatively, you could also change the background theme of your Developer Guide to a lighter colour scheme. Just a recommendation, would it be better for you to use a boolean "True" and "False" instead of "1" and "0"